### PR TITLE
dagger implementation 1

### PR DESCRIPTION
**Pull request by lkenigsb**

### CODE REVIEW

File: src/main/java/kenigsberg/transit/TranitServiceModule.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package kenigsberg.transit;
+
+import dagger.Module;
+import dagger.Provides;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+@Module
+public class TranitServiceModule {
```
Comment (andrewoid) [inline src/main/java/kenigsberg/transit/TranitServiceModule.java]: Transit

File: src/main/java/kenigsberg/transit/TransitComponent.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package kenigsberg.transit;
+
+import dagger.Component;
+
+import javax.inject.Singleton;
+
+public interface TransitComponent {
+    @Singleton
+    @Component(modules = {TranitServiceModule.class})
+    public interface ForecastWeatherComponent {
```
Comment (andrewoid) [inline src/main/java/kenigsberg/transit/TransitComponent.java]: 👀

File: src/main/java/kenigsberg/transit/TransitService.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package kenigsberg.transit;
 
+import dagger.Module;
```
Comment (andrewoid) [inline src/main/java/kenigsberg/transit/TransitService.java]: Unnecessary

File: src/main/java/kenigsberg/transit/TransitController.java [hunk 4]
```diff
@@ -19,6 +20,7 @@ public class TransitController {
     private DefaultTableModel tableModel;
 
 
+    @Inject
     public TransitController(TransitService service, JLabel limitedInfo,
                              DefaultTableModel tableModel) {
```
Comment (andrewoid) [inline src/main/java/kenigsberg/transit/TransitController.java]: The Frame needs to share the same JLabel and DefaultTableModel

**Comment by lkenigsb** [comment]
I dont know why when running it is now giving me an error message about the Provides annotation...

File: src/main/java/kenigsberg/transit/TransitFrame.java [hunk 5]
```diff
@@ -62,13 +77,13 @@ public TransitFrame() {
 
         String columnNames[] = {"Stop Name", "Arrival Time", "Distance"};
 
-        DefaultTableModel defaultTableModel = new DefaultTableModel(columnNames, 0);
+        defaultTableModel = new DefaultTableModel(columnNames, 0);
```
Comment (andrewoid) [inline src/main/java/kenigsberg/transit/TransitFrame.java]: You're supposed to use the defaultTableModel being passed into the constructor.

File: src/main/java/kenigsberg/transit/TransitServiceModule.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    @Singleton
+    public JLabel providesLimitedInfo()
+    {
+        return new JLabel();
+    }
+
+    @Provides
+    @Named("tableModel")
+    @Singleton
+    public JLabel providesTableModel()
```
Comment (andrewoid) [inline src/main/java/kenigsberg/transit/TransitServiceModule.java]: 👀

File: src/main/java/kenigsberg/transit/TransitServiceModule.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        return new JLabel();
+    }
+
+    @Provides
+    @Named("defaultTableModel")
+    @Singleton
+    public DefaultTableModel providesTableModel()
+    {
+        //String columnNames[] = {"Stop Name", "Arrival Time", "Distance"};
+        //return new DefaultTableModel(columnNames, 0);
```
Comment (andrewoid) [inline src/main/java/kenigsberg/transit/TransitServiceModule.java]: Why is this commented out?

File: src/main/java/kenigsberg/transit/TransitFrame.java [hunk 8]
```diff
@@ -21,11 +26,21 @@ public class TransitFrame extends JFrame {
...
     private TransitController controller;
+    private JLabel limitedInfoLabel;
 
-    public TransitFrame() {
+    private DefaultTableModel defaultTableModel;
+
+
+    @Inject
+    public TransitFrame(TransitController controller,
+                        @Named("limitedInfoLabel") JLabel limitedInfoLabel,
```
Comment (andrewoid) [inline src/main/java/kenigsberg/transit/TransitFrame.java]: You should be able to figure out this error.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**